test/Microsoft.IdentityModel.Tokens.Tests/EventBasedLRUCacheTests.cs

```diff
@@ -0,0 +1,242 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.CryptoProviderCacheOptions
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using Microsoft.IdentityModel.TestUtils;
+using Xunit;
+
+namespace Microsoft.IdentityModel.Tokens.Tests
+{
+    public class EventBasedLRUCacheTests
+    {
+        [Fact]
+        public void Contains()
+        {
+            TestUtilities.WriteHeader($"{this}.Contains");
+            var context = new CompareContext($"{this}.Contains");
+            var cache = new EventBasedLRUCache<int?, string>(10);
+
+            cache.SetValue(1, "one");
+            if (!cache.Contains(1))
+                context.AddDiff("Cache should contain the key value pair {1, 'one'}, but the Contains() method returned false.");
+
+            cache.TryRemove(1, out _);
+            if(cache.Contains(1))
+                context.AddDiff("The key value pair {1, 'one'} should have been removed from the cache, but the Contains() method returned true.");
+
+            try
+            {
+                cache.Contains(null);
+                context.AddDiff("The parameter passed into the Contains() method was null, but no exception was thrown.");
+            }
+            catch (Exception ex)
+            {
+                if (!ex.GetType().Equals(typeof(ArgumentNullException)))
+                    context.AddDiff("The exception type thrown by Contains(null) was not of type ArgumentNullException.");
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        [Fact]
+        public void RemoveExpiredValues()
+        {
+            TestUtilities.WriteHeader($"{this}.RemoveExpiredValues");
+            var context = new CompareContext($"{this}.RemoveExpiredValues");
+            var cache = new EventBasedLRUCache<int, string>(10);
+
+            for (int i = 0; i <= 10; i++)
+            {
+                // Only even values should expire.
+                if (i % 2 == 0)
+                    cache.SetValue(i, i.ToString(), DateTime.UtcNow + TimeSpan.FromSeconds(5));
+                else
+                    cache.SetValue(i, i.ToString());
+            }
+
+            Thread.Sleep(5000);
```

Comment:
>Thread.Sleep(5000); [](start = 12, length = 19)

see other comment on thread.sleep

Reply:
I see you have the method you can use instead now?

---
In reply to: [574028899](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1575#discussion_r574028899) [](ancestors = 574028899)